src/sentry_core.c

```diff
@@ -864,17 +864,21 @@ sentry_transaction_start_child(
         max_spans = options->max_spans;
     }
 
-    return sentry__start_child(max_spans, parent, operation, description);
-    // TODO: add pointer to transaction on child span
+    sentry_value_t span
+        = sentry__value_span_new(max_spans, parent, operation, description);
+    return sentry__span_new(opaque_parent, span);
```

Comment:
an unfortunate consequence of the change to `sentry__span_new` is that it does create some confusion for contributors(/developers). they'll need to understand the difference between `sentry__value_span_new` and `sentry__span_new`, and why and when one needs to be invoked over the other.